### PR TITLE
Add notification GitHub action workflow

### PR DESCRIPTION
**Pull request by manitgupta**
Adds a GitHub action workflow for generating notification on the specified conditions.  The notification destination is a custom web-hook which is stored in the Secrets Manager in the repo (Settings -> Secrets -> Actions).

Current conditions for notifications are:
1. Push to master.
2. Pull request raised to the repo.

### CODE REVIEW

**Comment by manitgupta** [comment]
Notification workflow is failing because PR checks are not able to read the secrets (WEBHOOK URL) stored in the repo until they are merged.

File: .github/workflows/chat-notification.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2020 Google LLC
```
Comment (Deep1998) [inline .github/workflows/chat-notification.yaml:1]: 2022 :)

**Review by Deep1998 — APPROVED** [review]
LGTM

File: .github/workflows/chat-notification.yaml (hunk 1 above)
Comment (manitgupta) [reply .github/workflows/chat-notification.yaml]: Fixed!

**Comment by manitgupta** [comment]
Notification workflow is failing because workflows inside a PR cannot access secrets of a repo.

**Pull request merged**